hrms/overrides/custom_employee_advance.py

```diff
@@ -0,0 +1,366 @@
+from hrms.hr.doctype.employee_advance.employee_advance import EmployeeAdvance
+from suite42_erp_app.overrides.custom_task import CustomTask, TaskType
+from erpnext.accounts.doctype.payment_entry.payment_entry import (
+    get_party_details,
+    get_account_details,
+)
+from suite42_erp_app.suite42_erp_app.doctype.suite42_application_config.suite42_application_config import (
+    Suite42ApplicationConfig,
+)
+from suite42_erp_app.suite42_erp_app.common.utils import user_has_role
+from erpnext.accounts.doctype.sales_invoice.sales_invoice import get_bank_cash_account
+from suite42_erp_app.suite42_erp_app.common.authentication.reimbursement_constants import (
+    RoleConstants,
+    EmployeeAdvanceConstants,
+    CompanyConstants,
+)
+from suite42_erp_app.overrides.custom_company.constants import CompanyNames
+from suite42_erp_app.suite42_erp_app.common.decorators import (
+    handle_exceptions_with_readable_message,
+)
+from frappe.utils import cstr, flt
+import frappe
+from frappe import _
+import json
+from datetime import datetime
+
+
+class CustomEmployeeAdvance(EmployeeAdvance):
+    def validate(self):
```

Comment:
If employee type is whitelisted only then allow in advance and expense claim

Reply:
Done